### PR TITLE
[FLINK-10136] [table] Add REPEAT supported in Table API and SQL

### PR DESCRIPTION
**Pull request by yanghua**
## What is the purpose of the change

*This pull request add REPEAT supported in Table API and SQL*

## Brief change log

  - *Add REPEAT supported in Table API and SQL*

## Verifying this change

This change is already covered by existing tests, such as *ScalarFunctionsTest#testRepeat*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / **docs** / JavaDocs / not documented)

### CODE REVIEW

**Comment by yanghua** [comment]
cc @xccui can you review this PR? Thanks.

**Comment by yanghua** [comment]
fixed checkstyle issue @xccui

**Comment by xccui** [comment]
Thanks for your work, @yanghua. The PR looks good to me. +1 to merge.

File: docs/dev/table/functions.md [hunk 1]
```diff
@@ -2614,6 +2626,18 @@ STRING.rtrim()
...
+
+    <tr>
+      <td>
+        {% highlight java %}
+STRING.repeat(INT)
+{% endhighlight %}
+      </td>
+      <td>
+        <p>Returns a string that repeats the base <i>STRING</i> <i>INT</i> times.</p> 
+        <p>E.g., <code>"This is a test String. ".repeat(2)</code> returns "This is a test String.This is a test String.".</p>
```
Comment (xccui) [inline docs/dev/table/functions.md]: There's an additional space after the full stop. (I've fixed it in my branch, so just let it be there.)

Comment (yanghua) [reply docs/dev/table/functions.md]: Thank you very much. @xccui

**Pull request closed without merging**